frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx

```diff
@@ -439,146 +439,132 @@ export const ProvidedAPIPage = connectToModel((props: ProvidedAPIPageProps) => {
-  // TODO(alecmerdler): Use additional `x-descriptor` to specify if should be considered main?
-  const isMainDescriptor = (descriptor: Descriptor) => {
-    return ((descriptor['x-descriptors'] as StatusCapability[]) || []).some((type) => {
-      switch (type) {
-        case StatusCapability.podStatuses:
-          return true;
-        default:
-          return false;
-      }
-    });
-  };
-
-  const blockValue = (descriptor: Descriptor, block: { [key: string]: any }) =>
-    !_.isEmpty(descriptor) ? _.get(block, descriptor.path, descriptor.value) : undefined;
-
-  const { kind, metadata, spec, status } = props.obj;
-
-  // Find the matching CRD spec for the kind of this resource in the CSV.
-  const ownedDefinitions = _.get(
-    props.clusterServiceVersion,
-    'spec.customresourcedefinitions.owned',
-    [],
-  );
-  const reqDefinitions = _.get(
-    props.clusterServiceVersion,
-    'spec.customresourcedefinitions.required',
-    [],
-  );
-  const thisDefinition = _.find(
-    ownedDefinitions.concat(reqDefinitions),
-    (def) => def.name.split('.')[0] === props.kindObj.plural,
-  );
-  const statusDescriptors = _.get<Descriptor[]>(thisDefinition, 'statusDescriptors', []).filter(
-    // exclude Conditions since they are included in their own section
-    (descriptor) => descriptor.path !== 'conditions',
-  );
-  const specDescriptors = _.get<Descriptor[]>(thisDefinition, 'specDescriptors', []);
-  const currentStatus = _.find(statusDescriptors, { displayName: 'Status' });
-  const primaryDescriptors = statusDescriptors.filter((descriptor) => isMainDescriptor(descriptor));
-
-  const header = (
-    <h2 className="co-section-heading">{thisDefinition?.displayName || kind} Overview</h2>
-  );
-
-  const primaryDescriptor = primaryDescriptors.map((statusDescriptor: Descriptor) => {
-    return (
-      <div key={statusDescriptor.displayName} className="col-sm-6">
-        <StatusDescriptor
-          descriptor={statusDescriptor}
-          value={blockValue(statusDescriptor, status)}
-          obj={props.obj}
-          model={props.kindObj}
-        />
-      </div>
-    );
-  });
-
-  const [errorMessage, setErrorMessage] = React.useState(null);
-  const handleError = (errorMsg: string) => setErrorMessage(errorMsg);
+const OperandDetailsSection: React.FC = ({ children }) => (
+  <div className="co-operand-details__section co-operand-details__section--info">
+    <div className="row">{children}</div>
+  </div>
+);
 
-  const details = (
-    <div className="co-operand-details__section co-operand-details__section--info">
-      <div className="row">
-        <div className="col-sm-6">
-          <ResourceSummary resource={props.obj} />
-        </div>
-        {currentStatus && (
-          <div className="col-sm-6" key={currentStatus.path}>
+const PodStatuses: React.FC<PodStatusesProps> = ({ kindObj, obj, podStatusDescriptors = [] }) =>
+  podStatusDescriptors.length > 0 ? (
```

Comment:
I'd use optional chaining instead defaulting with `= []`. It handles `null` rather just `undefined`.

```suggestion
  podStatusDescriptors?.length > 0 ? (
```